src/elm/Pages/Secrets/View.elm

```diff
@@ -429,6 +451,44 @@ updateSecretHref type_ secret =
                 Routes.SharedSecret "native" secret.org encodedTeam encodedName
 
 
+{-| copySecret : takes a secret and returns the yaml struct of the secret
+-}
+copySecret : Secret -> String
+copySecret secret =
+    let
+        yaml =
+            "- name: " ++ secret.name ++ "\n  key: " ++ secret.key ++ "\n  engine: native\n  type: "
+    in
+    case secret.type_ of
+        Vela.OrgSecret ->
+            yaml ++ "org"
+
+        Vela.RepoSecret ->
+            yaml ++ "repo"
+
+        Vela.SharedSecret ->
+            yaml ++ "shared"
+
+
+{-| copyButton : copy button that copys secret yaml to clipboard
+-}
+copyButton : String -> Html Msg
+copyButton copyYaml =
+    button
+        [ class "copy-button"
+        , attribute "aria-label" <| "copy secret yaml '" ++ copyYaml ++ "' to clipboard "
```

Comment:
i think we should simplify the `aria-label` text. i think when screen readers read this aloud it will not be super helpful after "copy secret yaml". maybe just leave out the `copyYaml` part?